src/viztracer/decorator.py

```diff
@@ -97,7 +106,16 @@ def wrapper(*args, **kwargs) -> Any:
                 "dur": dur,
                 "cat": "FEE",
             }
-            tracer.add_raw(raw_data)
+            local_tracer.add_raw(raw_data)
             return ret
 
         return wrapper
+
+
+def _log_sparse_wrapper_with_args(stack_depth: int = 0, dynamic_tracer_check: bool = False) -> Callable:
```

Comment:
This function is only used once and the content is a single line. I don't see a very strong reason to abstract it out. I think we can just return the `partial` when the `func` is `None` in `log_sparse`.